File: leetcode/java/MergeIntervals.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class MergeIntervals{
    public int[][] merge(int[][] intervals){
        //ascending order of first element
        Arrays.sort(intervals, (i,j)-> i[0]-j[0]);
        List<int[]> answer = new ArrayList<>();
        for(int[] interval : intervals){
            if(answer.isEmpty() || answer.getLast()[1] < interval[0]){
                answer.add(interval);
            }else{
                answer.getLast()[1] = Math.max(answer.getLast()[1], interval[1]);
            }
        }
        return answer.toArray(new int[answer.size()][]);
    }
}

File: leetcode/cpp/merge-intervals.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());
        vector<vector<int>> answer;
        for(auto nextInterval : intervals){
            if(answer.size()==0 || answer.back()[1] < nextInterval[0]){
                answer.push_back(nextInterval);
            }
            else{
                answer.back()[1] = max(answer.back()[1], nextInterval[1]);
            }
        }
        return answer;
    }
};